### PR TITLE
add tox file

### PR DESCRIPTION
**Pull request by nv-alaiacano**
Adds a `tox.ini` file to execute jenkins/gpu-based tests in a controlled venv.

Site packages from the base container are included in the `test-gpu` testenv, with `requirements/dev.txt` plus the latest commits to merlin-core installed on top of it.

We need to update the jenkins config to run `tox -er test-gpu` instead of `./ci/test_unit.sh` and ensure that the environment variables there are used.

Fixes: https://github.com/NVIDIA-Merlin/models/issues/793

### CODE REVIEW

**Comment by nv-alaiacano** [comment]
rerun tests

**Comment by nv-alaiacano** [comment]
rerun tests

**Comment by nv-alaiacano** [comment]
rerun tests

**Comment by nv-alaiacano** [comment]
rerun tests

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+envlist = python3.8,test-gpu
+
+[testenv]
+commands =
+    pip install --upgrade pip
+    pip install -e .[all]
+
+[testenv:test-gpu]
+passenv=
+    CUDA_VISIBLE_DEVICES
```
Comment (edknv) [inline tox.ini:14]: Do we need this environment at all? It looks like it's set to `CUDA_VISIBLE_DEVICES=0,1` in Jenkins, but Models doesn't have mult-gpu support yet, and horovod, which we plan to use as the backed for distirubted training in Models, discourages the use of this environment variable. I'm wondering if existing tests still work if we remove it from Jenkins.

**Review by oliverholworthy — APPROVED** [review]

**Review by marcromeyn — APPROVED** [review]

**Comment by edknv** [comment]
rerun tests

**Pull request merged**